src/controllers/userController.js

```diff
@@ -7,66 +7,63 @@ const {
 } = require('../errors');
 
 const getUserDetails = async (req, res, next) => {
-  const user = await User.findById(req.params.id);
-  if (!user) {
-    throw new NotFoundError(`No user with id ${req.params.id}`);
-  } else {
-    res.status(StatusCodes.OK).json({ user });
-  }
+  res.status(StatusCodes.OK).json({ user: req.user });
 };
 
 const updatePassword = async (req, res, next) => {
-  const user = await User.findById(req.params.id).select('+password');
-  const isPasswordCorrect = await user.comparePassword(req.body.oldPassword);
-  if (!user) {
-    throw new NotFoundError(`No user with id ${req.params.id}`);
-  } else {
-    if (!isPasswordCorrect) {
-      throw new UnauthenticatedError('Invalid Password');
-    }
-    if (req.body.newPassword !== req.body.confirmPassword) {
-      throw new UnauthenticatedError('Password does not match');
-    }
-    user.password = req.body.newPassword;
-    await user.save();
-    res.status(StatusCodes.OK).json({
-      msg: `Password for user ${user.name} was successfully updated `,
-    });
+  const { oldPassword, newPassword, confirmPassword } = req.body;
+  console.log(oldPassword, newPassword, confirmPassword);
+  if (!oldPassword || !newPassword || !confirmPassword) {
```

Comment:
Usually it is client side concern if passwords match, but if you do go with this approach, you should let your frontend team that they need to send both values.